client/components/foldable-card/index.jsx

```diff
@@ -72,86 +91,79 @@ class FoldableCard extends Component {
-			const iconSize = 24;
-			const screenReaderText = this.props.screenReaderText || this.props.translate( 'More' );
-			return (
-				<button
-					disabled={ this.props.disabled }
-					type="button"
-					className="foldable-card__action foldable-card__expand"
-					onClick={ clickAction }
-				>
-					<ScreenReaderText>{ screenReaderText }</ScreenReaderText>
-					<Gridicon icon={ this.props.icon } size={ iconSize } />
-				</button>
-			);
-		}
-	}
-
-	renderContent() {
-		return <div className="foldable-card__content">{ this.props.children }</div>;
-	}
-
-	renderHeader() {
-		const summary = this.props.summary ? (
-			<span className="foldable-card__summary">{ this.props.summary } </span>
-		) : null;
-		const expandedSummary = this.props.expandedSummary ? (
-			<span className="foldable-card__summary-expanded">{ this.props.expandedSummary } </span>
-		) : null;
-		const headerClickAction = this.props.clickableHeader ? this.getClickAction() : null;
-		const headerClasses = classNames( 'foldable-card__header', {
-			'is-clickable': !! this.props.clickableHeader,
-			'has-border': !! this.props.summary,
-		} );
-		return (
-			<div className={ headerClasses } onClick={ headerClickAction }>
-				<span className="foldable-card__main">{ this.props.header } </span>
-				<span className="foldable-card__secondary">
-					{ summary }
-					{ expandedSummary }
-					{ this.renderActionButton() }
-				</span>
-			</div>
-		);
-	}
-
 	render() {
-		const Container = this.props.compact ? CompactCard : Card;
-		const itemSiteClasses = classNames( 'foldable-card', this.props.className, {
-			'is-disabled': !! this.props.disabled,
+		const {
+			actionButton,
+			actionButtonExpanded,
+			className,
+			clickableHeader,
+			compact,
+			disabled,
+			expandedSummary,
+			header,
+			icon,
+			screenReaderText,
+			summary,
+			translate,
+		} = this.props;
+
+		const Container = compact ? CompactCard : Card;
+
+		const itemSiteClasses = classNames( 'foldable-card', className, {
+			'is-disabled': !! disabled,
 			'is-expanded': !! this.state.expanded,
-			'has-expanded-summary': !! this.props.expandedSummary,
+			'has-expanded-summary': !! expandedSummary,
+		} );
+
+		const headerClasses = classNames( 'foldable-card__header', {
+			'is-clickable': !! clickableHeader,
+			'has-border': !! summary,
 		} );
 
 		return (
 			<Container className={ itemSiteClasses }>
-				{ this.renderHeader() }
-				{ this.state.expanded && this.renderContent() }
+				<div
+					className={ headerClasses }
+					onKeyPress={ this.onHeaderKeyPress }
+					onClick={ this.clickHeader }
+					role="button"
+					tabIndex={ 0 }
+				>
+					<span className="foldable-card__main">{ header } </span>
+					<span className="foldable-card__secondary">
+						{ summary && <span className="foldable-card__summary">{ summary } </span> }
+						{ expandedSummary && (
+							<span className="foldable-card__summary-expanded">{ expandedSummary } </span>
+						) }
+						{ actionButton ? (
```

Comment:
It looks like `actionButton` is not used correctly throughout Calypso, after a quick search through it looks like the places that use `actionButton` pass a button component (plugins, adding a contact form), so maybe this should be updated to match general use? (that can be another issue/PR, just wanted to note it somewhere)